Add information about certificate bundles/chains (#2590)
sensu/sensu-docs: content/sensu-go/5.21/reference/agent.md

Configuration via command line flags overrides attributes specified in a configuration file.
See the [Example Sensu agent configuration file][5] for flags and defaults.

### Certificate bundles or chains

> The Sensu backend supports all types of certificate bundles (or chains) as long as the server (or leaf) certificate is the *first* certificate in the bundle.

Reviewer: I'm assuming this should mention "the sensu agent" instead of "the sensu backend" given that it's the agent reference?

Reviewer: ditto with "the server (or leaf) certificate"; I think that should read "the agent (or leaf) certificate".

> The Sensu backend supports all types of certificate bundles (or chains) as long as the server (or leaf) certificate is the *first* certificate in the bundle.

Author: ```suggestion
The Sensu agent supports all types of certificate bundles (or chains) as long as the agent (or leaf) certificate is the *first* certificate in the bundle.
```

> This is because the Go standard library assumes that the first certificate listed in the PEM file is the server certificate &mdash; the certificate that the program will use to show its own identity.

Reviewer: I would say "leaf certificate" instead of "server certificate" since we're talking about the Go standard library here, and leaf is a better term since it doesn't have to be a server specifically.

> This is because the Go standard library assumes that the first certificate listed in the PEM file is the server certificate &mdash; the certificate that the program will use to show its own identity.

Author: ```suggestion
This is because the Go standard library assumes that the first certificate listed in the PEM file is the leaf certificate &mdash; the certificate that the program will use to show its own identity.
```

> If you send the server certificate alone instead of sending the whole bundle with the server certificate first, you will see a `certificate not signed by trusted authority` error.

Reviewer: Again, I think it should mention "agent" instead of "server" here.

> If you send the server certificate alone instead of sending the whole bundle with the server certificate first, you will see a `certificate not signed by trusted authority` error.

Author: ```suggestion
If you send the leaf certificate alone instead of sending the whole bundle with the leaf certificate first, you will see a `certificate not signed by trusted authority` error.
```

> You must present the whole chain to the remote so it can determine whether it trusts the server certificate through the chain.

Reviewer: "presented certificate" instead of "server certificate"

> You must present the whole chain to the remote so it can determine whether it trusts the server certificate through the chain.

Author: ```suggestion
You must present the whole chain to the remote so it can determine whether it trusts the presented certificate through the chain.
```

| cert-file  |      |
-------------|------

> description  | Path to the agent certificate file used in mutual TLS authentication. Sensu supports certificate bundles (or chains) as long as the server (or leaf) certificate is the *first* certificate in the bundle.

Author: ```suggestion
description  | Path to the agent certificate file used in mutual TLS authentication. Sensu supports certificate bundles (or chains) as long as the agent (or leaf) certificate is the *first* certificate in the bundle.
```
